File: 0313A.Llya.and.Bank.Account/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

func Fprintf(format string, x ...interface{}) {
	fmt.Fprintf(writer, format, x...)
}

func Fscanf(format string, x ...interface{}) {
	fmt.Fscanf(reader, format, x...)
}

func Solution() {
	var balance int
	Fscanf("%d\n", &balance)

	if balance >= 0 {
		Fprintf("%d\n", balance)
	} else {
		balance = -balance
		if balance < 10 {
			Fprintf("0\n")
		} else {
			lastDigit := balance % 10
			digits := (balance - lastDigit) / 10
			prevDigit := digits % 10

			var result int
			if lastDigit < prevDigit {
				result = -((digits - prevDigit) + lastDigit)
			} else {
				result = -((digits - prevDigit) + prevDigit)
			}

			Fprintf("%d\n", result)
		}
	}
}

func main() {
	Solution()
	writer.Flush()
}

File: 0313A.Llya.and.Bank.Account/solution.py
def solution():
	balance = int(input())

	if balance >= 0:
		print(balance)
		return

	balance = -balance
	if balance < 10:
		print("0")
		return

	last_digit = balance % 10
	digits = (balance - last_digit) // 10
	prev_digit = digits % 10

	if last_digit < prev_digit:
		result = -((digits - prev_digit) + last_digit)
	else:
		result = -((digits - prev_digit) + prev_digit)

	print(result)


if __name__ == "__main__":
	solution()
